### DISCUSSION TITLE
"back to search results" button?

### DISCUSSION THREAD

**Discussion by merchgirl** [NightscoutFoundation/xDrip#2072] [Ideas]
I log all of my treatments, and I like using the note search feature to look back on the last few times I ate a certain meal, how much I dosed, and what happened to my levels. When I search for a keyword and it brings up the last few entries that contain it, I generally want to look at more than one result in order to compare, but after I click on one of them, I need to restart the search (enter the keywords and dates to search) again. I'd love for there to be a "back to search results" button so that I don't have to start a new search each time. 
Anyone else agree? Wonder if this might be worth starting a feature request or if it's just me that wishes this existed.

**Comment by Navid200** [comment]
@tolot27 
@tzachi-dar 
@jamorham

**Comment by Navid200** [comment]
Hopefully one of the individuals I have tagged knows what you mean and will respond.

But, if you could add screenshots so that for example, I could repeat what you explain so that I can see it happen, it may help.  
I don't use treatments.  So, I have no idea how to do any of it.  I have no idea how to search the treatment notes ...

**Comment by merchgirl** [comment]
Once you use the dropper in the upper right of the screen to enter a treatment (either carbs or insulin), if you enter a description for said treatment (using the note with a plus on it that's right under the dropper), it will be saved alongside that carb or insulin entry as an attached note. You can search your notes by going to the hamburger menu > note search and entering your search term(s) and desired dates. It will display a list of results, and if you click one, it will bring you to that date in your history. I'd love there to then be a "back to search results" button on that next screen so that I could return to the list of results instead of having to start over to look at the next result down. Hitting the back button on my phone takes me to the xdrip home screen. Let me know if these screenshots are helpful or if there's anything else you'd like to see.
![Screenshot_20220410-171232](https://user-images.githubusercontent.com/90476025/162646538-b0afd297-028c-485d-87f9-5411e17327ac.jpg)
![Screenshot_20220410-171632](https://user-images.githubusercontent.com/90476025/162646547-a3caf64c-30aa-4612-8a28-cf09fc4c5db6.jpg)

**Reply by Navid200** [reply]
Isn't every content of every note visible after you enter the search item?

For example in the above image where you have entered chia as your search term, is there anything in any of the notes that you cannot see already?

Why do you need to tap on any of them?

**Comment by Navid200** [comment]
I see.

Please go to "Issues" above.  Tap/click on "New issue"."
Then, click on "Get started" in front of "Feature request".

Enter the description there.
You need a developer who understand what you want and can do it.  You have no control over what developers can do.  But, you have control on how well you can describe the request so that it becomes easy for anyone reading the description to understand it.
If the developer who can do it never understands what you want, it will be a shame.

Reading your description, it sounds like you enter the note exactly at the same time as you enter the insulin or carb value.  But, when I tried it, it only worked for me when I entered the insulin first.  Then, clicked on it and clicked on "ADD NOTE" at the bottom.  

Anyway, I can see that using Android navigation back button just leaves everything and does not take you back to the search results.  So, I can see how useful what you are asking can be.  But, this is beyond my capability to add.  We have developers who can do this though.
So, please go ahead and open a feature request as I described.
Please let me know if you have any questions.

**Comment by merchgirl** [comment]
Hi! Sorry I dropped the ball on this! My notes would say something like
"chia cereal, bolused 3u 50/50 over 45 mins" and the next result might say
"chia cereal, bolused 3.5u 60/40 over 45 mins" and I liked to click on the
individual notes to go to that point on my graph so I could see how each
worked out and what I might want to change to eat that same thing again, so
a back button to the list of results would make it easier than retyping the
search time and changing time constraints in order to look at the graph for
the next result. But before I managed to start an issue for a feature
request, I wound up getting an iPhone and starting on a DIY loop system
(iAPS) so I'm no longer using the xdrip notes feature.
